temporian/core/operators/calendar/base.py

```diff
@@ -56,9 +66,32 @@ def build_op_definition(cls) -> pb.OperatorDef:
         return pb.OperatorDef(
             key=cls.operator_def_key(),
             inputs=[pb.OperatorDef.Input(key="sampling")],
+            attributes=[
+                pb.OperatorDef.Attribute(
+                    key="utc_offset",
+                    type=pb.OperatorDef.Attribute.Type.FLOAT_64,
+                ),
+            ],
             outputs=[pb.OperatorDef.Output(key="output")],
         )
 
+    @staticmethod
+    def convert_to_utc_offset(timezone: Union[str, float, int]) -> float:
+        if isinstance(timezone, str):
+            tz_time = datetime.datetime.now(tz=pytz.timezone(timezone))
+            offset_timedelta = datetime.datetime.utcoffset(tz_time)
+            return (
+                0.0
+                if offset_timedelta is None
+                else offset_timedelta.total_seconds() / 3600
+            )
+        return float(timezone)
```

Comment:
add case for timezone not being one of the accepted types (since runtime type checking can be disabled by the user I don't think we want to rely on it heavily)